src/Stack/Types/Config.hs

```diff
@@ -441,6 +443,21 @@ instance FromJSON ApplyGhcOptions where
             "everything" -> pure AGOEverything
             _ -> fail $ "Invalid ApplyGhcOptions: " ++ show t
 
+data AllowNewer
+  = AllowNewerAll
+  | AllowNewerNone
+  | AllowNewerPackages [String]
+  deriving (Show, Read, Eq, Ord)
+
+instance FromJSON AllowNewer where
```

Comment:
I am wondering if what is needed is `WithJSONWarnings AllowNewer` instance of `FromJSON`.